boards/toradex/verdin_imx8mm/doc/index.rst

```diff
@@ -0,0 +1,268 @@
+`i.MX 8M Applications Processor Reference Manual`_  (section 2.1.2 and 2.1.3)
+
+At compilation time you have to choose which RAM will be used. This
+configuration is done in the file ``boards/arm/verdin_imx8mm_m4/verdin_imx8mm_m4.dts``
+with "zephyr,flash" (when CONFIG_XIP=y) and "zephyr,sram" properties.
+The available configurations are:
+
+.. code-block:: none
+
+   "zephyr,flash"
+   - &tcml_code
+   - &ocram_code
+   - &ocram_s_code
+
+   "zephyr,sram"
+   - &tcmu_sys
+   - &ocram_sys
+   - &ocram_s_sys
+
+Starting the Cortex-M4 via U-Boot
+=================================
+
+Load and run Zephyr on M4 from A53 using u-boot by copying the compiled
+``zephyr.bin`` to the eMMC (can be the FAT or EXT4 partition). You can do it
+by using a USB stick or through the ethernet with the scp command, for example.
+Power it up and stop at the u-boot prompt.
+
+Load the M4 binary onto the desired memory and start its execution using:
+
+.. code-block:: console
+
+   fatload mmc 0:1 ${loadaddr} zephyr.bin
+   cp.b ${loadaddr} 0x7e0000 <size_of_binary_in_bytes>
+   bootaux 0x7e0000
+
+Or if the binary is on the ext4 partition:
+
+.. code-block:: console
+
+   ext4load mmc 0:2 ${loadaddr} /path/to/zephyr.bin
+   cp.b ${loadaddr} 0x7e0000 <size_of_binary_in_bytes>
+   bootaux 0x7e0000
+
+If you are using `TorizonCore`_ OS, then you should use partition 1:
+
+.. code-block:: console
+
+   ext4load mmc 0:1 ${loadaddr} /path/to/zephyr.bin
+   cp.b ${loadaddr} 0x7e0000 <size_of_binary_in_bytes>
+   bootaux 0x7e0000
+
+Starting the Cortex-M4 via RemoteProc
+=====================================
+
+Copy the ``zepyhr.elf`` to ``/lib/firmware`` on the target.
+
+.. note::
+   In order to use remoteproc you have to add ``imx8mm-verdin_hmp_overlay.dtbo`` at
+   the end of the line in the ``/boot/overlays.txt``, then reboot the target. If
+   you are using `TorizonCore`_, then this file is located at
+   ``/boot/ostree/torizon-<hash>/dtb/overlays.txt``.
+
+To load and start a firmware use these commands:
+
+.. code-block:: console
+
+   verdin-imx8mm:~# echo zepyhr.elf > /sys/class/remoteproc/remoteproc0/firmware
+   verdin-imx8mm:~# echo start > /sys/class/remoteproc/remoteproc0/state
+   [   94.714498] remoteproc remoteproc0: powering up imx-rproc
+   [   94.720481] remoteproc remoteproc0: Booting fw image zephyr.elf, size 473172
+   [   94.727713] remoteproc remoteproc0: No resource table in elf
+   [   94.733615] remoteproc remoteproc0: remote processor imx-rproc is now up
+
+The M4-Core is now started up and running. You can see the output from Zephyr
+on UART4.
+
+Debugging
+=========
+
+MIMX8MM EVK board can be debugged by connecting an external JLink
+JTAG debugger to the J902 debug connector and to the PC. Then
+the application can be debugged using the usual way.
+
+Here is an example for the :ref:`hello_world` application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: verdin_imx8mm_m4
```

Comment:
Board target should be `verdin_imx8mm/mimx8mm6/m4`

Reply:
This has been addressed in the latest push.